ote/src/cljc/ote/geo.cljc

```diff
@@ -174,30 +137,34 @@
 
 
 
-;;; Conversion between WGS84 and Finnish EUREF-FIN systems
+;; Conversion from KKJ to WGS84 (needed for Kalkati station positions)
 
 #?(:clj
-   (def euref-wkt "PROJCS[\"EUREF_FIN_TM35FIN\", \n  GEOGCS[\"GCS_EUREF_FIN\", \n    DATUM[\"D_ETRS_1989\", \n      SPHEROID[\"GRS_1980\", 6378137.0, 298.257222101]], \n    PRIMEM[\"Greenwich\", 0.0], \n    UNIT[\"degree\", 0.017453292519943295], \n    AXIS[\"Longitude\", EAST], \n    AXIS[\"Latitude\", NORTH]], \n  PROJECTION[\"Transverse_Mercator\"], \n  PARAMETER[\"central_meridian\", 27.0], \n  PARAMETER[\"latitude_of_origin\", 0.0], \n  PARAMETER[\"scale_factor\", 0.9996], \n  PARAMETER[\"false_easting\", 500000.0], \n  PARAMETER[\"false_northing\", 0.0], \n  UNIT[\"m\", 1.0], \n  AXIS[\"x\", EAST], \n  AXIS[\"y\", NORTH]]"))
-
-#?(:clj (def wgs84 org.geotools.referencing.crs.DefaultGeographicCRS/WGS84))
-#?(:clj (def euref (org.geotools.referencing.CRS/parseWKT euref-wkt)))
-#?(:clj (def euref->wgs84-transform (org.geotools.referencing.CRS/findMathTransform euref wgs84 true)))
-#?(:clj (def swap-coordinates
-          (reify com.vividsolutions.jts.geom.CoordinateFilter
-            (filter [_ c]
-              (let [x (.-x c)
-                    y (.-y c)]
-                (set! (.-x c) y)
-                (set! (.-y c) x))))))
+   ;; We need to define our own KKJ CRS because the one defined in geotools
+   ;; has the wrong bursa wolf parameters (the TOWGS84 line below) that resulted
+   ;; in wrong coordinates for kalkati stations (~10 - 15 meters off).
+   (def kkj
+          (org.geotools.referencing.CRS/parseWKT "PROJCS[\"KKJ / Finland Uniform Coordinate System\",
+GEOGCS[\"KKJ\",DATUM[\"Kartastokoordinaattijarjestelma_1966\",
+SPHEROID[\"International 1924\",6378388,297,AUTHORITY[\"EPSG\",\"7022\"]],
+TOWGS84[-90.7, -106.1, -119.2, 4.09, 0.218, -1.05, 1.37],
```

Comment:
We should mention the tool where to get them in case a similar need arises again: http://www.epsg-registry.org/